taxi/src/main/java/com/epam/training/taxi/Main.java

```diff
@@ -0,0 +1,24 @@
+package com.epam.training.taxi;
+
+import com.epam.training.taxi.account.Account;
+import com.epam.training.taxi.config.BillingServiceConfiguration;
+import com.epam.training.taxi.invoice.Invoice;
+import com.epam.training.taxi.service.BillingService;
+
+import java.io.IOException;
+
+public class Main {
+
+    public static void main(String[] args) throws IOException {
+        Long accountId = Long.valueOf(args[0]);
+        Double distanceTravelled = Double.valueOf(args[1]);
+
+        BillingService billingService = BillingServiceConfiguration.getBillingService();
+
+        Account userAccount = billingService.getAccount(accountId);
+
+        Invoice invoice = billingService.createInvoice(userAccount, distanceTravelled);
```

Comment:
By providing the result of the [getAccount method](https://github.com/epam-deik-cooperation/epam-deik-java-dev/blob/29043a42aee97c2e9c1a4410462fc99d612ddee1/taxi/src/main/java/com/epam/training/taxi/Main.java#L18) call to another method of the same service class, means, there is a leaky abstraction here.
It could be avoided by calling `getAccount(accountId)` method from the  `createInvoice(` method. In this case, the  `createInvoice(` method should get the `accountId` as a method parameter instead of the `userAccount`.
For example: billingService.createInvoice(accountId, distanceTravelled); 

Articles about leaky abstraction:
- https://devopedia.org/leaky-abstractions#qst-ans-0
- https://medium.com/young-coder/what-are-leaky-abstractions-an-illustrated-guide-f2982ff21cae
- http://www.principles-wiki.net/principles:law_of_leaky_abstractions